### PR TITLE
Add inference API of AMR

### PR DESCRIPTION
**Pull request by h-munakata**
# Summary
1. Add inference API of AMR
  - Add msclap in dependencies
  - Add configuration for AMR in ```lighthouse/models.py```
   I implemented ```encode_audio()``` separately from ```encode_video()```.
```
model.encode_audio("api_example/1a-ODBWMUAE.wav")
```

2. Modularize PANNs and CLAP
  - ```Class AudioEncoder``` in ```lighthouse/feature_extractor/audio_encoder.py``` only have model selector and ```lighthouse/feature_extractor/audio_encoders/{clap_a | pann}.py``` has individual model.

# Future work
- Add Gradio demo

### CODE REVIEW

**Comment by h-munakata** [comment]
The test has failed... I'm checking.

File: setup.py [hunk 1]
```diff
@@ -5,6 +5,6 @@
     version='0.1',
     install_requires=['easydict', 'pandas', 'tqdm', 'pyyaml', 'scikit-learn', 'ffmpeg-python',
                       'ftfy', 'regex', 'einops', 'fvcore', 'gradio', 'torchlibrosa', 'librosa',
-                      'clip@git+https://github.com/openai/CLIP.git'],
+                      'clip@git+https://github.com/openai/CLIP.git', 'msclap'],
```
Comment (awkrail) [inline setup.py:8]: Could you add msclap before 'clip@git+https://github.com/openai/CLIP.git' ?

File: training/evaluate.py [hunk 2]
```diff
@@ -439,7 +439,7 @@ def check_valid_combination(dataset, feature, domain):
     is_valid = check_valid_combination(args.dataset, args.feature, args.domain)
 
     if is_valid:
-        option_manager = BaseOptions(args.model, args.dataset, args.feature, args.domain)
+        option_manager = BaseOptions(args.model, args.dataset, args.feature, False, args.domain)
```
Comment (awkrail) [inline training/evaluate.py:442]: Could you remove magic number of False? Instead, please insert variable X = False and then use it here for readability?

**Comment by awkrail** [comment]
@h-munakata In addition, no tests are newly provided. Hence, could you add some tests for CLAP feature extractor and AMR inference API?

**Comment by h-munakata** [comment]
> @h-munakata In addition, no tests are newly provided. Hence, could you add some tests for CLAP feature extractor and AMR inference API?

I see. I will add some tests in ```tests/test_models.py``` later.

**Comment by h-munakata** [comment]
For the test and the Gradio demo, I think {moment | qd | cg | ...}-detr for CLAP features is useful.
Should I prepare the pre-trained weights of all seven models trained with Clotho-Moment?

**Comment by awkrail** [comment]
> For the test and the Gradio demo, I think {moment | qd | cg | ...}-detr for CLAP features is useful.
Should I prepare the pre-trained weights of all seven models trained with Clotho-Moment?

Yes, I agree. Could you prepare the pre-trained weights on your Zenodo?

**Comment by h-munakata** [comment]
> Yes, I agree. Could you prepare the pre-trained weights on your Zenodo?
Sure. I will upload after the training.

**Comment by awkrail** [comment]
@h-munakata Sorry, I misunderstood your question. I think that in your paper, CG-DETR (or QD-DETR) achieved the highest performance. So, no need for training. All you need to do is upload the current trained models on Zenodo and accessible from Lighthouse.

**Comment by h-munakata** [comment]
My intention in training all models is to add a CLAP feature to the double for loop of FEATURE and MODEL used in the demo and test to make it easier to handle.
As you said, I will stop training all models and define AUDIO_FEATURE variable in the demo and test.

>All you need to do is upload the current trained models on Zenodo and accessible from Lighthouse.

I see. I'll upload the model in the next commit for the test.

**Comment by awkrail** [comment]
@h-munakata BTW, could you finish implementing web demo tomorrow? I will tag the current version as v1.0, and wondering whether you can finish this implementation by tomorrow.

**Comment by h-munakata** [comment]
> BTW, could you finish implementing web demo tomorrow? 
Yes, I want to make it in time for the DCASE workshop the day after tomorrow.

**Pull request merged**